### ISSUE TITLE
Allow overriding database host when using `withContainer` (Docker-in-Docker)

### ISSUE THREAD

**Issue by joschi** [monosoul/jooq-gradle-plugin#119]
When using the plugin with a container (`jooq.withContainer`), it currently doesn't allow overriding the database host used to build the JDBC URL.

Some CI setups utilizing Docker-in-Docker (DinD) require overriding the hostname and use something other than `localhost` due to the fact that the Docker daemon is running _outside_ of the actual container.

In our specific case, we're using GitLab CI/CD and the CI runners are Docker containers themselves with access to the Docker daemon running on the host machines.

Overriding the database host depending on where the build is running would enable us to use the same build logic (except for the different host name) locally and on the CI runners.

**Comment by monosoul** [comment]
Hey @joschi , when running with container, the plugin uses `testcontainers`' API to get the hostname of the container to connect to. Testcontainers provide a way to configure host override: https://www.testcontainers.org/features/configuration/#customizing-docker-host-detection

Altho, since `testcontainers` lib is shadowed by the plugin, I had to adjust the property names to avoid conflicts with a non-shadowed lib (if there is one in the classpath).

So, to sum this up: try to set `DEV_MONOSOUL_JOOQ_TESTCONTAINERS_HOST_OVERRIDE` env variable in you CI env to the hostname you need. I think that should do the trick for you.

Please, let me know if that works (or if it doesn't).
